Shared/ThresholdKeyView.swift

```diff
@@ -148,23 +165,14 @@ struct ThresholdKeyView: View {
                                     }
                                     shareCount += 1
                                 } while !finishedFetch
-
                                 // There are 0 locally available shares for this tkey
                                 if shareCount == 0 {
-                                    // Attempt reconstruction
-                                    guard let reconstructionDetails = try? await threshold_key.reconstruct() else {
-                                        alertContent = "Failed to reconstruct key. \(threshold) more share(s) required"
-                                        resetAccount = true
-                                        showAlert = true
-                                        showSpinner = SpinnerLocation.nowhere
-                                        return
-                                    }
-
-                                    // save shares up to required threshold
                                     let shareIndexes = try threshold_key.get_shares_indexes()
 
-                                    // TODO: Save only one share, which is device share and not all shares.
-                                    for i in 0..<threshold {
+                                    for i in 0..<shareIndexes.count {
+                                        if shareIndexes[Int(i)] == "1" {
+                                            continue
```

Comment:
So it will skip share "1" and save all others here right? This is not a single share.

Reply:
no. as you can see in line 195, if single share is saved, for loop will be broken by `break`

Reply:
Ok, so then why the need for the loop if it is only going to execute once?

Why not just exclude the index of "1" from shareIndexes array and save the first one in the resulting next?

One way to filter out a value in an array is like this:
```swift
   var shareIndexes = try threshold_key.get_shares_indexes()
   shareIndexes.removeAll(where: {$0 == "1"})
```

Reply:
good suggestion. I'll apply this.